vizro-core/examples/kpi/utils/_charts.py

```diff
@@ -0,0 +1,198 @@
+"""Contains custom components and charts used inside the dashboard."""
+
+from typing import List, Literal, Optional
+
+import dash_bootstrap_components as dbc
+import pandas as pd
+import vizro.models as vm
+import vizro.plotly.express as px
+from dash import html
+from vizro.models.types import capture
+
+
+# CUSTOM COMPONENTS -------------------------------------------------------------
+# Note: This is a static KPI Card only (it will not be reactive to controls). A new dynamic KPI Card component
+# is currently in development.
+class KPI(vm.VizroBaseModel):
+    """Static custom `KPI` Card."""
+
+    type: Literal["kpi"] = "kpi"
+    title: str
+    value: str
+    icon: str
+    sign: Literal["delta-pos", "delta-neg"]
+    ref_value: str
+
+    def build(self):
+        return dbc.Card(
+            [
+                html.H4(self.title),
+                html.P(self.value),
+                html.Span(
+                    [
+                        html.Span(self.icon, className="material-symbols-outlined"),
+                        html.Span(self.ref_value),
+                    ],
+                    className=self.sign,
+                ),
+            ],
+            className="kpi-card-ref",
+        )
+
+
+# CUSTOM CHARTS ----------------------------------------------------------------
+@capture("graph")
+def bar(  # noqa: PLR0913
+    x: str,
+    y: str,
+    data_frame: pd.DataFrame,
+    top_n: int = 15,
+    color_discrete_sequence: Optional[List[str]] = None,
+    custom_data: Optional[List[str]] = None,
+):
+    df_agg = data_frame.groupby([y]).aggregate({x: "count"}).sort_values(by=x, ascending=False).reset_index()
```

Comment:
```suggestion
    df_agg = data_frame.groupby(y).agg({x: "count"}).sort_values(by=x, ascending=False).reset_index()
```
I think should work the same.

The alias `df.agg` is also recommended over `df.aggregate` (works exactly the same).